src/Streamly/Internal/Data/Stream/IsStream/Top.hs

```diff
@@ -356,23 +370,34 @@ leftJoin eq s1 s2 = Stream.evalStateT (return False) $ do
 --
 -- Time: O(m + n)
 --
--- /Unimplemented/
+-- /Pre-release/
 {-# INLINE hashLeftJoin #-}
-hashLeftJoin :: -- Hashable b =>
-    (a -> b -> Bool) -> t m a -> t m b -> t m (a, Maybe b)
-hashLeftJoin = undefined
+hashLeftJoin ::  (IsStream t, Ord a, Monad m, Monad (t m)) =>
+    t m a -> t m a -> t m (a, Maybe a)
+hashLeftJoin  s1 s2 =
+    Stream.concatM $ do
+        l2 <- Stream.fold (Fold.classify Fold.toList) 
+            $ StreamK.adapt 
+            $ fmap (, ()) s2
+        let res = do
+                e <- s1
+                if e `Map.member` l2
+                then  return (e, Just e)
+                else  return (e, Nothing)
+        return res
 
 -- | Like 'leftJoin' but works only on sorted streams.
 --
 -- Space: O(1)
 --
 -- Time: O(m + n)
 --
--- /Unimplemented/
+-- /Pre-release/
 {-# INLINE mergeLeftJoin #-}
-mergeLeftJoin :: -- Monad m =>
+mergeLeftJoin :: forall (t :: (Type -> Type) -> Type -> Type) m a b.
+    (IsStream t, MonadIO m, Eq a, Eq b) =>
     (a -> b -> Ordering) -> t m a -> t m b -> t m (a, Maybe b)
-mergeLeftJoin _eq _s1 _s2 = undefined
+mergeLeftJoin  = Stream.leftJoinByMerge
```

Comment:
Let's keep the name of the function same in this module as well as in the `Stream` module.

Reply:
This applies to all such names in this module.